Tests/Client/Unit Tests/AtomicTests.swift

```diff
@@ -0,0 +1,234 @@
+
+    func test_Atomic_whenValueChanged_callsDidSetCallback() {
+        // Setup
+        let initialValue = "Luke"
+        let newValue = "Leia"
+
+        var callbackNewValue: String?
+        var callbackOldValue: String?
+
+        let value = Atomic(initialValue, callbackQueue: .testQueue) {
+            XCTAssertTrue(DispatchQueue.isTestQueue, "Callback called on an incorrect queue.")
+            callbackNewValue = $0
+            callbackOldValue = $1
+        }
+        assert(value.get() == initialValue)
+
+        // Action
+        value.set(newValue)
+
+        // Assert
+        AssertEqualEventually(callbackNewValue, newValue)
+        AssertEqualEventually(callbackOldValue, initialValue)
+    }
+
+    func test_Atomic_keyPathHelpers() {
+        // Setup
+        struct Helper: Equatable {
+            var elements: [Int] = []
+        }
+        let value = Atomic(Helper())
+
+        // Actions
+        value.update(\.elements, to: [0])
+        XCTAssertEqual(value.get(), Helper(elements: [0]))
+
+        value.elements = [1]
+        XCTAssertEqual(value.get(), Helper(elements: [1]))
+
+        value.append(to: \.elements, 2)
+        XCTAssertEqual(value.get(), Helper(elements: [1, 2]))
+    }
+
+    func test_Atomic_dictionaryHelpers() {
+        let value = Atomic(["Luke": 1])
+        XCTAssertEqual(value["Luke"], 1)
+    }
+
+    func test_Atomic_whenCalledRecursively_worksCorrectly() {
+        // Setup
+        let value = Atomic("Luke")
+
+        // Action
+        value.update { (oldValue) -> String? in
+            // The current value should be same as `oldValue`
+            XCTAssertEqual(value.get(), oldValue)
+
+            value.set("Skywalker")
+            // The current value should be updated
+            XCTAssertEqual(value.get(), "Skywalker")
+
+            value.update { oldValue in
+                // The current value should be same as `oldValue`
+                XCTAssertEqual(value.get(), oldValue)
+                return "Vader!"
+            }
+            // The current value should be updated
+            XCTAssertEqual(value.get(), "Vader!")
+
+            // Finally, the return value from `update` closure should override previous changes
+            return "Leia"
+        }
+
+        XCTAssertEqual(value.get(), "Leia")
+    }
+}
+
+// MARK: - Stress tests
+
+extension AtomicTests {
+
+    var numberOfStressTestCycles: Int { 50 }
+
+    // Uncomment the test below to run the full stress test suite.
+    // Don't forget to increase `numberOfStressTestCycles` to some more interesting value.
+
+//    func test_Atomic_underHeavyLoad_behavesCorrently() {
+//        for _ in 0..<100 {
+//            test_Atomic_usedAsCounter_behavesCorrectly()
+//            test_Atomic_usedWithCollection_behavesCorrectly()
+//            test_Atomic_whenSetAndGetCalledSimultaneously_behavesCorrectly()
+//            test_Atomic_whenCalledFromMainThred_behavesCorrectly()
+//        }
+//    }
```

Comment:
Why do we need to keep this test commented?

Reply:
Because it can take tens of seconds to run. We don't need to run the stress tests every time if nothing has changed in `Atomic`. We still run the tests, but only once and for `numberOfStressTestCycles: 50` so obvious regressions will be caught.

`test_Atomic_underHeavyLoad_behavesCorrently` is meant to be used only if you changed something in `Atomic` and you want to be sure it behaves correctly under extreme load.

Reply:
Got it! Maybe we can add a comment on why they should be kept disabled?

Reply:
Actually, I just found out it's ok to have it uncommented if `numberOfStressTestCycles` is this small. I'll change the documentation to reflect this.